devito/ops/operator.py

```diff
@@ -54,13 +55,22 @@ def _specialize_iet(self, iet, **kwargs):
         # be generated (since a set is an unordered collection)
         to_dat = filter_sorted(to_dat)
 
+        iteration_tree = retrieve_iteration_tree(iet, mode='normal')[0]
```

Comment:
[0] ? 
what if there's none?

Reply:
I added a test in case there is none. 

But basically,if there is none, means that there is no loop and then there is no code to be specialized using OPS.

Reply:
yeah, what I meant was that instead of adding a test, you could `return iet` and perhaps raise a warning in case there's none. 

Devito can generate codes beyond finite differences, which might not fit the OPS abstraction, so...

Reply:
Good idea, I'll do it.

Reply:
Fixed.

I also moved this verification to the beginning of the `specialize_iet` because there is no point in continuing the specialization for OPS if there is no iteration tree.